src/server/game/Entities/AreaTrigger/AreaTrigger.cpp

```diff
@@ -100,14 +148,256 @@ void AreaTrigger::Update(uint32 p_time)
         Remove(); // expired
 
     WorldObject::Update(p_time);
+    _timeSinceCreated += p_time;
+
+    sScriptMgr->OnAreaTriggerEntityUpdate(this, p_time);
+
+    switch (GetTemplate()->GetType())
+    {
+        case AREATRIGGER_TYPE_SPHERE:
+            SearchUnitInSphere();
+            break;
+        case AREATRIGGER_TYPE_BOX:
+            SearchUnitInBox();
+            break;
+        case AREATRIGGER_TYPE_POLYGON:
+            SearchUnitInPolygon();
+            break;
+        case AREATRIGGER_TYPE_CYLINDER:
+            SearchUnitInCylinder();
+            break;
+    }
 }
 
 void AreaTrigger::Remove()
 {
     if (IsInWorld())
     {
+        if (Unit* caster = GetCaster())
+            caster->_UnregisterAreaTrigger(this);
+
+        for (ObjectGuid insideUnitGuid : _insideUnits)
+            if (Unit* leavingUnit = ObjectAccessor::GetUnit(*this, insideUnitGuid))
+                sScriptMgr->OnAreaTriggerEntityUnitExit(this, leavingUnit);
+
+        sScriptMgr->OnAreaTriggerEntityRemove(this);
+
         SendObjectDeSpawnAnim(GetGUID());
         RemoveFromWorld();
         AddObjectToRemoveList();
     }
 }
+
+void AreaTrigger::SearchUnitInSphere()
+{
+    std::list<Unit*> targetList;
```

Comment:
Yes you are right, leave it for now as it is, the searchers probably need an overall refactoring...